src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,1323 @@
+        return calibration_data
+
+    def _stopCalibration(self, *args):
+        """
+        Call stop calibration loop on the ASM and unsubscribe _startCalibration from all the VA's so the calibration
+        loop is not restarted when a VA is changed.
+
+        *args is need to support self subscribing
+        """
+        # Unsubscribe the descanner VA's
+        descanner = self._mirror_descanner
+        descanner.scanGain.unsubscribe(self._startCalibration)
+        descanner.rotation.unsubscribe(self._startCalibration)
+        descanner.scanOffset.unsubscribe(self._startCalibration)
+
+        # Unsubscribe the scanner VA's
+        scanner = self._ebeam_scanner
+        scanner.scanGain.unsubscribe(self._startCalibration)
+        scanner.dwellTime.unsubscribe(self._startCalibration)
+        scanner.rotation.unsubscribe(self._startCalibration)
+        scanner.scanDelay.unsubscribe(self._startCalibration)
+        scanner.scanOffset.unsubscribe(self._startCalibration)
+
+        # Unsubscribe the mppc VA
+        self._mppc.cellCompleteResolution.unsubscribe(self._startCalibration)
+
+        self._calibrationParameters = None
+        self.asmApiPostCall("/scan/stop_calibration_loop", 204)
+
+    def _setCalibrationMode(self, mode):
+        """
+        Setter for the calibration mode, this method calls the _startCalibration and _stopCalibration methods
+        (responsible for starting/stopping the calibration nd subscriptions to VA's to ensure updated parameters),
+        changes the default scanner and descanner gain values for calibration, and unsubscribes all the listeners
+        from the mppc.dataflow.
+
+        :param mode (bool):
+        """
+        if mode:
+            if self._mppc.data._count_listeners() != 0:
+                logging.warning("Unsubscribing all the listeners from the mppc.dataflow, calibration mode cannot be "
+                                "started if the dataflow still has subscribers.")
+
+                while self._mppc.data._count_listeners() != 0:
+                    try:
+                        self._mppc.data._listeners.pop()
+                    except:
+                        # If removing one of the listeners fails that is probably due to a lack of listeners
+                        break
+
+            # Standard values of scan gain for calibration mode
+            self._mirror_descanner.scanGain.value = (1.0, 0.0)
+            self._ebeam_scanner.scanGain.value = (1.0, 1.0)
+            self._startCalibration()  # Start calibration loop and subscribe to VA's to reload updated parameters
+            return True
+
+        else:
+            self._stopCalibration()  # Stop calibration loop and unsubscribe from all the VA's
+            # Standard values of scan gain for acquisition mode
+            self._mirror_descanner.scanGain.value = (1.0, 1.0)
+            self._ebeam_scanner.scanGain.value = (1.0, 1.0)
```

Comment:
This is the same value as in calibrationMode = True. So why set it at all?

Reply:
No here all the gain values are set to 1 (for calibarition = True the mirro descanner y value is set to zero) I assumed 1 as gain is a good default value after returning from the calibration mode.
But maybe later we want to change that or use persistent data for that.

Reply:
I wonder if it isn't better to put back the previous value for the gain instead of a new default value?

Reply:
We could do that indeed also, do you like to have that also implemented?
That would require though to also save that value before setting the default calibration values.

Reply:
Yes I think that makes sense. And it is just one extra line of code to save the previous values.

Reply:
Since we don't change these values anymore at all this issue is obsolete isn't it @Sabrina85 ?

Reply:
yes!